Backend/ChatAPI.Startup/Program.cs

```diff
@@ -7,6 +12,19 @@
 builder.Services.AddEndpointsApiExplorer();
 builder.Services.AddSwaggerGen();
 
+//custom configuration of settings
+builder.Services.Configure<PersistenceSettings>(builder.Configuration.GetSection("connectionString"));
+//Db register
+builder.Services.AddDbContext<ChatAppDbContext>(o => o.UseNpgsql(@builder.Configuration.GetSection("PersistenceSettings")["connectionString"], 
```

Comment:
This is not going to read connection string. The selected node doesnt have a proprtty that named as "ConnectionString".

You have to use your class that you defined for binding the settings to it. 

Inject an instance of PersistenceSettings class like that;
      var serviceProvider = services.BuildServiceProvider();
      var providerOptions = serviceProvider.GetService<IOptions<PersistenceSettings >>()?.Value ??
                              throw new InvalidOperationException("The persistence provider options cannot be null.");
                              
And then read it like that;                              
providerOptions.GetConnectionString()